ansible/roles/vm_set/tasks/remove_topo.yml

```diff
@@ -51,23 +51,22 @@
       loop_var: dut_name
 
   - name: Unbind topology {{ topo }} to VMs. base vm = {{ VM_base }}
-    vm_topology:
-      cmd: "unbind"
-      vm_set_name: "{{ vm_set_name }}"
-      topo: "{{ topology }}"
-      vm_names: "{{ VM_hosts }}"
-      vm_base: "{{ VM_base }}"
-      vm_type: "{{ vm_type }}"
-      duts_fp_ports: "{{ duts_fp_ports }}"
-      duts_midplane_ports: "{{ duts_midplane_ports }}"
-      duts_inband_ports: "{{ duts_inband_ports }}"
-      duts_mgmt_port: "{{ duts_mgmt_port }}"
-      duts_name: "{{ duts_name.split(',') }}"
-      max_fp_num: "{{ max_fp_num }}"
-      dut_interfaces: "{{ dut_interfaces | default('') }}"
-      is_vs_chassis: "{{ is_vs_chassis | default(false) }}"
-      batch_mode: "{{ batch_mode if batch_mode is defined else omit }}"
+    include_tasks: unbind_vm.yml
+    loop: "{{ VM_hosts | flatten(levels=1) }}"
+    loop_control:
+      loop_var: vm_name
+
+  - name: Wait for unbind tasks to complete
     become: yes
+    async_status:
+      jid: "{{ async_unbind_job_results[vm_name] }}"
+    loop: "{{ VM_hosts | flatten(levels=1) }}"
+    loop_control:
+      loop_var: vm_name
```

Comment:
From here, `vm_name` sounds like an item of `vm_hosts`. Then should it be string type?

Reply:
Actually we can, but I want to keep pace with the argument `vm_names`